### PR TITLE
KEYCLOAK-15109 Ingress/Route Host Override

### PR DESCRIPTION
**Pull request by jeesmon**
Support overriding default host set in Ingress/Route through Keycloak CR

## JIRA ID
https://issues.redhat.com/browse/KEYCLOAK-15109

## Checklist:
- [x] Automated Tests
- [ ] [Documentation](https://github.com/keycloak/keycloak-documentation) changes if necessary
- [ ] (If required) Discussed on [Keycloak DEV](https://groups.google.com/forum/#!forum/keycloak-dev)

### CODE REVIEW

**Review by slaskawi — CHANGES_REQUESTED** [review]
Thanks for the Pull Request @jeesmon !

Unfortunately, I'm not keen on merging this change. Our Operator supports both Routes and Ingresses depending what types are available in the cluster. The `host` argument is only respected when using Ingress and completely ignored for Routes. This may be misleading to other users and therefore, can not be merged as is.

I can think of a few solutions here:
- Leave the behavior as is. Note that the Ingress `host` parameter is never reconciled. You can alter it to whatever you want and the Operator will not override it.
- Implement the missing part of the Routes. [See the OpenShift manual for Routes](https://docs.openshift.com/container-platform/3.9/architecture/networking/routes.html#route-hostnames).
- Create a nasted part of the spec that is dedicated only to Ingresses. Something like: `spec.externalAccess.ingress.host`.

I'm definitely in favor of the second solution mentioned here (implement the missing part for the Routes).

**Comment by jeesmon** [comment]
Thanks @slaskawi  I will look into option 2 and update PR

**Comment by jeesmon** [comment]
@slaskawi Updated PR with support host override for Route. Could you please review? Thanks

**Comment by slaskawi** [comment]
Dear contributor,

I'm putting this Pull Request on hold until we get our CI back (see https://github.com/keycloak/keycloak-operator/pull/266). Once the Github Actions Pull Request is merged, I will kindly ask you for rebasing your changes. Then, we'll proceed with a normal code review process.

Please see [this Keycloak DEV thread](https://groups.google.com/g/keycloak-dev/c/ijI9grpRw74) to learn more why we are migrating to Github Actions.

Sorry for inconvenience

**Review by slaskawi — APPROVED** [review]

**Comment by slaskawi** [comment]
Integrated. Thanks @jeesmon !

May I also kindly ask you to add a short note on the new functionality here: https://github.com/keycloak/keycloak-documentation/blob/master/server_installation/topics/operator/keycloak-cr.adoc ?

**Comment by jeesmon** [comment]
Will do @slaskawi

**Comment by jeesmon** [comment]
@slaskawi https://github.com/keycloak/keycloak-documentation/pull/1058

File: deploy/crds/keycloak.org_keycloaks_crd.yaml
Comment (dhx-teuto) [inline deploy/crds/keycloak.org_keycloaks_crd.yaml:64]: This description differs from the one in `keycloak_types.go`. Maybe `make code/gen` again?
I know it's already merged but I just spotted it while rebasing my own pull request.

Comment (jeesmon) [reply deploy/crds/keycloak.org_keycloaks_crd.yaml:64]: Created new PR to fix the mismatch. Thanks a lot for reporting the mismatch @dhx-teuto. Really appreciated.

**Pull request merged**
